Marco d'Itri wrote:

I run one of the debian mirrors, and I had to write this patch because
my archive is split between more than one disk. Would you accept a more
polished version of this patch for inclusion in rsync?

To use this patch, run these commands for a successful build:

    patch -p1 <patches/soften-links.diff
    ./configure                           (optional if already run)
    make

based-on: fe2ef556d9ef11e5dd549e19a06a7a924f7ddfa1
diff --git a/syscall.c b/syscall.c
--- a/syscall.c
+++ b/syscall.c
@@ -123,9 +123,14 @@ ssize_t do_readlink(const char *path, char *buf, size_t bufsiz)
 #ifdef HAVE_LINK
 int do_link(const char *old_path, const char *new_path)
 {
+	int st;
+
 	if (dry_run) return 0;
 	RETURN_ERROR_IF_RO_OR_LO;
-	return link(old_path, new_path);
+	st = link(old_path, new_path);
+	if (/*soften_links &&*/ st != 0 && errno == EXDEV)
+		st = symlink(old_path, new_path);
+	return st;
 }
 #endif
 
